src/amp/components/Experiment.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+export interface ExperimentModel {
+    [key: string]: {
+        sticky: boolean;
+        consentNotificationId?: string;
+        variants: {
+            [key: string]: number; // We do this for compatibility with ExperimentModel via ExperimentConfig
+        };
+    };
+}
+
+interface StyledVariants { [key: string]: { proportion: number; style: string;}; }
+
+export interface ExperimentFullConfig {
+    [key: string]: {
+        sticky: boolean;
+        consentNotificationId?: string;
+        variants: StyledVariants;
+    };
+}
+
+
+export interface ExperimentStyle {
+    [key: string]: {[key: string]: string};
+}
+
+const extractStyleFromVariants = (variants: StyledVariants): [{[key: string]: number}, {[key: string]: string}] =>
+    Object.entries(variants).reduce(
+    (acc, [variant, config])=>{
+        const {proportion, style} = config;
+        acc = [{...acc[0], ...{[variant]: proportion}}, {...acc[1], ...{[variant]: style}}];
+        return acc
+    },
+    [{}, {}]
+);
+export const extractModelAndStyle = (fullConfig: ExperimentFullConfig): [ExperimentModel, ExperimentStyle] => {
+    return Object.entries(fullConfig).reduce(
+        (extracted, [experimentName, styledModel])=>{
+            const {variants, ...configWithoutVariants} = styledModel;
+            const [proportions, styles] = extractStyleFromVariants(variants);
+            extracted[0][experimentName] = {...configWithoutVariants, ...{variants: proportions}};
+            extracted[1][experimentName] = styles;
+            return extracted
+        },
+        [{}, {}] as [ExperimentModel, ExperimentStyle])
+};
+
+export const getActiveTests = (abTestObject: ExperimentFullConfig,
+                               switches: {[key: string]: boolean}): ExperimentFullConfig => {
+    return Object.entries(abTestObject).reduce(
+        (acc, [testName, testConfig]) => {
+            if (switches[testName]) {
+                acc[testName] = testConfig;
+            }
+            return acc;
+        },
+        {} as ExperimentFullConfig
+    );
+}
+
+
+export const buildExperimentStyle = (experiments: ExperimentStyle): string => {
+    return Object.entries(experiments).reduce(
+        (experimentStyleString, [experimentName, styleVariants]) => {
+            const experimentStyle = Object.entries(styleVariants).reduce(
+                (variantStyleString, [variantName, variantStyle]) =>{
+                    return `${variantStyleString} body[amp-x-${experimentName}='${variantName}'] ${variantStyle}`
+                }
+                ,
+                ''
+            );
+            return `${experimentStyleString} ${experimentStyle}`
+        },
+        ''
+    )
+};
+
+export const Experiment: React.FC<{
+    experiment: ExperimentModel;
+}> = experiment => {
+    const script: string = `<amp-experiment>
+        <script type="application/json">
+        ${JSON.stringify(experiment.experiment)}
```

Comment:
Where's experiment.experiment? When I look at the example data  ( ab-zero-test-experiment ), or the ExperimentModel type (well, interface)  I can't see an 'experiment' key, so I don't understand where this comes from.

Reply:
For some reason the object passed through to experiment was being put inside another object where the key was the prop (here that's `experiment`). I'm certain there's a way of accessing it that's better than what I've used but I couldn't figure it out myself.

Reply:
Improved this to use idiomatic React code with destructuring of the props argument.